src/modm/driver/pressure/ms5837_impl.hpp

```diff
@@ -0,0 +1,154 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2023, Vivien Henry
+ * Based on the ms5837, courtesy of Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_MS5837_HPP
+#   error "Don't include this file directly, use 'ms5837.hpp' instead!"
+#endif
+
+#include <modm/math/utils/endianness.hpp>
+
+namespace modm
+{
+
+template < typename I2cMaster >
+Ms5837<I2cMaster>::Ms5837(DataBase &data, uint8_t address) : I2cDevice<I2cMaster, 2>(address), data(data)
+{
+}
+
+// -----------------------------------------------------------------------------
+
+template < typename I2cMaster >
+modm::ResumableResult<uint16_t>
+Ms5837<I2cMaster>::initialize()
+{
+    RF_BEGIN();
+
+    // Reset sensor once after power-on to load calibration PROM into the internal registers
+    buffer[0] = i(Command::Reset);
+    this->transaction.configureWrite(buffer, 1);
+    if (RF_CALL( this->runTransaction() ))
+    {
+        MODM_LOG_DEBUG.printf("MS5837 Reset Success\n");
+    }
+    else {
+        MODM_LOG_DEBUG.printf("MS5837 Reset Failed\n");
+        RF_RETURN(false);
+    }
+
+
+    // 4 ms reload (?: from MS5611)
+    timeout.restart(std::chrono::milliseconds(4));
+    RF_WAIT_UNTIL(timeout.isExpired());
+
+    // Read the factory calibration from PROM
+    data.prom.data[0] = modm::fromBigEndian(RF_CALL(readProm(0)));
+    data.prom.data[1] = modm::fromBigEndian(RF_CALL(readProm(1)));
+    data.prom.data[2] = modm::fromBigEndian(RF_CALL(readProm(2)));
+    data.prom.data[3] = modm::fromBigEndian(RF_CALL(readProm(3)));
+    data.prom.data[4] = modm::fromBigEndian(RF_CALL(readProm(4)));
+    data.prom.data[5] = modm::fromBigEndian(RF_CALL(readProm(5)));
+    data.prom.data[6] = modm::fromBigEndian(RF_CALL(readProm(6)));
+    // data.prom.data[7] = modm::fromBigEndian(RF_CALL(readProm(7)));
+
+    MODM_LOG_DEBUG.printf("MS5837 Reset Initialize done\n");
+
+    RF_END_RETURN(data.prom.calculateCrc());
+}
+
+// -----------------------------------------------------------------------------
+
+template < typename I2cMaster >
+modm::ResumableResult<void>
+Ms5837<I2cMaster>::readout(OversamplingRatio osrPressure, OversamplingRatio osrTemperature)
+{
+    RF_BEGIN();
+
+    // start a pressure conversion
+    buffer[0] = i(Command::Convert) | i(Conversion::Pressure) | i(osrPressure);
+    this->transaction.configureWrite(buffer, 1);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Pressure Conversion failed\n");
+        RF_RETURN();
```

Comment:
```suggestion
        RF_RETURN(false);
```
If you change the return type to `modm::ResumableResult<bool>` you can externalize the logger info.
Besides the log messages seem to be copy-pasted and not reflect the actual failed action ;-P

Reply:
ok you got me on this one